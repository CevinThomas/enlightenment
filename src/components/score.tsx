import React, {useEffect, useState} from 'react';
import {StyleSheet, Text, TouchableOpacity, View} from "react-native";
import FadeIn from "./fadeIn";
import GlobalStyles from "../utils/globalStyles"


const Score = (props) => {

    const [percentageCorrect, setPercentageCorrect] = useState(0);
    const [correctAttempts, setCorrectAnswers] = useState<number>(0);
    const [wrongAttempts, setWrongAttempts] = useState<number>(0);
    const [categoryAnswers, setCategoryAnswers] = useState<[]>([]);

    useEffect(() => {
        setCorrectAnswers(props.results.correctAnswers.length);
        setWrongAttempts(props.results.wrongAnswers.length);
        setCategoryAnswers(props.categoryAnswers);
        setPercentageCorrect(
            (props.firstTry / props.totalQuestions) * 100
        );
    }, []);

    return (
        <View style={styles.container}>
            <FadeIn>
                <View style={styles.headingContainer}>
                    <Text style={styles.mainheading}>Well done! This is how you did!</Text>
                </View>

                <View style={styles.textContainer}>
                    <Text style={styles.titles}>Total amount of questions: {props.totalQuestions}</Text>
                </View>

                <View style={styles.textContainer}>
                    <Text style={styles.titles}>Correct attempts: {correctAttempts}</Text>
                </View>

                <View style={styles.textContainer}>
                    <Text style={styles.titles}>Wrong attempts: {wrongAttempts}</Text>
                </View>

                <View style={styles.textContainer}>
                    <Text
                        style={styles.titles}>Percentage: {percentageCorrect}%</Text>
                </View>

                <View style={styles.textContainer}>
                    <Text style={[styles.titles, {marginBottom: 10}]}>Categories used:</Text>

                    {categoryAnswers.map(category => {
                        return (
                            <View key={category.name}>
                                <Text style={styles.titles}>{category.name} Scored: {category.timesCorrect} out
                                    of {category.totalQuestions}</Text>
                            </View>
                        );
                    })}
                </View>


                <TouchableOpacity onPress={() => props.navigation.navigate("Home")} style={styles.buttonContainer}>
                    <Text style={{color: GlobalStyles.darkColor, textAlign: "center"}}>Do it
                        again!
                    </Text>
                </TouchableOpacity>
            </FadeIn>
        </View>
    );
};

const styles = StyleSheet.create({
    textContainer: {
        backgroundColor: "white",
        marginTop: 15,
        padding: 15,
        borderRadius: 10,
        shadowOffset: {width: 0, height: 0},
        shadowColor: "#000",
        elevation: 100,
        shadowRadius: 5,
        shadowOpacity: 0.1

    },
    container: {
        padding: 20
    },
    headingContainer: {
        borderBottomWidth: 1,
        borderBottomColor: GlobalStyles.darkColor
    },
    mainheading: {
        fontSize: 18,
        color: GlobalStyles.darkColor,
        textAlign: "center",
        marginBottom: 20
    },
    titles: {
        color: GlobalStyles.darkColor,
    },
    buttonContainer: {
        marginTop: 20,
        backgroundColor: GlobalStyles.lightColor,
        borderRadius: 10,
        padding: 10,
        textAlign: "center",
        shadowOffset: {width: 0, height: 0},
        shadowColor: "#000",
        elevation: 100,
        shadowRadius: 5,
        shadowOpacity: 0.1
    }
});

export default Score;
