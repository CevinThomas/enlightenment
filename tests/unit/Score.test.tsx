import 'react-native';
import React from 'react';
import Score from '../../src/components/score';

import renderer from 'react-test-renderer';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

test('Score Component Renders', () => {
    const tree = renderer.create(
        <Score/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
