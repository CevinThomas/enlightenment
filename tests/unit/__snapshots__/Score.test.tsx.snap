// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score Component Renders 1`] = `
<View
  style={
    Object {
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#000",
          "borderBottomWidth": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 18,
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
      >
        Well done! This is how you did!
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 100,
          "marginTop": 15,
          "padding": 15,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
          }
        }
      >
        Total amount of questions: 
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 100,
          "marginTop": 15,
          "padding": 15,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
          }
        }
      >
        Correct attempts: 
        0
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 100,
          "marginTop": 15,
          "padding": 15,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
          }
        }
      >
        Wrong attempts: 
        0
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 100,
          "marginTop": 15,
          "padding": 15,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
          }
        }
      >
        Percentage: 
        0
        %
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 10,
          "elevation": 100,
          "marginTop": 15,
          "padding": 15,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 5,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {
              "marginBottom": 10,
            },
          ]
        }
      >
        Categories used:
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 10,
          "elevation": 100,
          "marginTop": 20,
          "opacity": 1,
          "padding": 10,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 5,
          "textAlign": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "textAlign": "center",
          }
        }
      >
        Do it again!
      </Text>
    </View>
  </View>
</View>
`;
